src/components/AuthPanel/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { Container, Link, Typography } from "@mui/material"
+import { Stack } from "@mui/system"
+import { navigate, useLocation } from "@reach/router"
+import { auth } from "../../index"
+import { useEffect } from "react"
+import { useAuthState } from "react-firebase-hooks/auth"
+import SplitLayout from "../../layouts/split"
+import { ChildType, LocationProps } from "../../types"
+import Illustration from "./Illustration"
+import { CenteredStack as Centered, Column, Image } from "./styles"
+
+const LeftColumn = Column
+const RightColumn = Column
+
+type AuthPanelProps = {
+  children: ChildType // Should have a single child component
+}
+
+const AuthPanel = ({ children }: AuthPanelProps) => {
+  const [user, authorizing] = useAuthState(auth)
+  const location = useLocation() as LocationProps
+
+  useEffect(() => {
+    const from = location?.state?.from || "/"
+    if (user) navigate(from, { replace: true })
```

Comment:
This is copied from the old codebase so good question but as far as I can tell it's redirecting the user to the home '/' once they're signed in or to the page that they came from. Eg. if you try and access an authenticated route and it redirected you to sign in then it will take you back there once signed in... I could be wrong though